src/resolvers/git.cr

```diff
@@ -425,7 +425,11 @@ module Shards
 
       output = capture ? IO::Memory.new : Process::Redirect::Close
       error = IO::Memory.new
-      status = Process.run(command, shell: true, output: output, error: error)
+      if path
+        status = Process.run(command, shell: true, output: output, error: error, chdir: path)
+      else
+        status = Process.run(command, shell: true, output: output, error: error)
+      end
```

Comment:
You should be able to directly pass a nilable value for `chdir` and `Process.run` whil change the directory when present.
```suggestion
      status = Process.run(command, shell: true, output: output, error: error, chdir: path)
```